### PR TITLE
chore(sn_node): skip section_tree update earlier for outdated sap

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

File: sn_interface/src/network_knowledge/section_tree/mod.rs [hunk 1]
```diff
@@ -217,47 +217,47 @@ impl SectionTree {
...
-                    )));
-                }
+
+            // Verify the new SAP with the SAP we know and the proof chain provided.
+            if !proof_chain.has_key(&sap.section_key()) {
+                // This case may happen when both the sender and receiver is about to using
+                // a new SAP. The AE-Update was sent before sender switching to use new SAP,
+                // hence it only contains proof_chain covering the old SAP.
+                // When the update arrives after the receiver got switched to use new SAP,
+                // this error will be complained.
```
Comment (oetyng) [inline sn_interface/src/network_knowledge/section_tree/mod.rs:243]: I'm a bit unsure what this means.

File: sn_interface/src/network_knowledge/section_tree/mod.rs [hunk 2]
```diff
@@ -275,13 +275,6 @@ impl SectionTree {
                 signed_sap.value
             )));
         }
-        // Make sure the proof chain can be trusted,
-        // i.e. check each key is signed by its parent/predecessor key.
-        if !proof_chain.self_verify() {
```
Comment (joshuef) [inline sn_interface/src/network_knowledge/section_tree/mod.rs:280]: Do we not still need this verification?

File: sn_interface/src/network_knowledge/sections_dag.rs [hunk 3]
```diff
@@ -301,7 +301,9 @@ impl SectionsDAG {
     /// other
     pub fn merge(&mut self, mut other: Self) -> Result<()> {
         if !other.self_verify() {
-            return Err(Error::InvalidSignature);
+            return Err(Error::UntrustedProofChain(format!(
+                "Proof chain failed self verification: {other:?}",
```
Comment (joshuef) [inline sn_interface/src/network_knowledge/sections_dag.rs:305]: Can we add a specific error variant here 🙇

**Review by joshuef — CHANGES_REQUESTED** [review]

File: sn_interface/src/network_knowledge/sections_dag.rs (hunk 3 above)
Comment (maqi) [reply sn_interface/src/network_knowledge/sections_dag.rs:305]: that `UntrustedProofChain` is already bit specific?
or what do you suggest to use?

Comment (joshuef) [reply sn_interface/src/network_knowledge/sections_dag.rs:305]: Not to use a string in an existing error, but rather specific error for this situation. (Ideally we should breakdown UntrustedProofChain into the actual varieties it has, but i know that's not really to do with this PR)

**Pull request merged**